backends/nxp/tests/test_batch_norm_fusion.py

```diff
@@ -0,0 +1,211 @@
+        eps: float = 1e-5,
+    ):
+        super().__init__()
+        self.linear = torch.nn.Linear(fc_in_features, fc_out_features, bias=bias)
+        self.batch_norm = BatchNormModule(input_rank, fc_out_features, eps)
+        self.eval()
+
+    def forward(self, x):
+        x = self.linear(x)
+        return self.batch_norm(x)
+
+
+@pytest.mark.parametrize(
+    "bias", [True, False], ids=lambda x: "Bias" if x else "No bias"
+)
+@pytest.mark.parametrize(
+    "input_shape", [[4, 6, 8], [2, 4, 6, 8]], ids=lambda x: f"{len(x)}D"
+)
+def test_batch_norm_conv_fusing(bias: bool, input_shape: list[int]):
+    example_input = (torch.ones(*input_shape),)
+
+    module = ConvBatchNormModule(bias, len(input_shape), 4)
+    program = torch.export.export_for_training(module, example_input, strict=True)
+    og_module = program.module()
+
+    pm = NeutronAtenPassManager()
+    graph_module_out = pm(deepcopy(program.module())).graph_module
+
+    # Make sure the fusion worked.
+    og_nodes = list(program.graph.nodes)
+    transformed_nodes = list(graph_module_out.graph.nodes)
+
+    assert len(og_nodes) == (11 if bias else 10)
+    assert og_nodes[9 if bias else 8].target.__name__ == "batch_norm.default"
+
+    assert len(transformed_nodes) == 5
+    assert not any(
+        node.op == "call_function" and "batch_norm" in node.target.__name__
+        for node in transformed_nodes
+    )
+
+    # Verify that the behavior has not changed.
+    input_data = torch.randn(input_shape, dtype=torch.float32)
+    out1 = og_module(input_data).detach().numpy()
+    out2 = graph_module_out(input_data).detach().numpy()
+    assert np.allclose(out1, out2, atol=3.0e-7)
+
+
+@pytest.mark.parametrize(
+    "bias", [True, False], ids=lambda x: "Bias" if x else "No bias"
+)
+def test_batch_norm_linear_fusing(bias: bool):
+    input_shape = (2, 4, 6, 8)
+    example_input = (torch.ones(*input_shape),)
+
+    module = LinearBatchNormModule(bias, 4, input_shape[-1], input_shape[1])
+    program = torch.export.export_for_training(module, example_input, strict=True)
+    og_module = program.module()
+
+    pm = NeutronAtenPassManager()
+    graph_module_out = pm(deepcopy(program.module())).graph_module
+
+    # Make sure the fusion worked.
+    og_nodes = list(og_module.graph.nodes)
+    transformed_nodes = list(graph_module_out.graph.nodes)
+
+    assert len(og_nodes) == (11 if bias else 10)
+    assert og_nodes[8 if bias else 7].target.__name__ == "linear.default"
+
+    assert len(transformed_nodes) == 5
+    assert not any(
+        node.op == "call_function" and "batch_norm" in node.target.__name__
+        for node in transformed_nodes
+    )
+
+    # Verify that the behavior has not changed.
+    input_data = torch.randn(input_shape, dtype=torch.float32)
+    out1 = og_module(input_data).detach().numpy()
+    out2 = graph_module_out(input_data).detach().numpy()
+    assert np.allclose(out1, out2, atol=1.2e-7)
+
+
+# TODO - enable within PR #10196
```

Comment:
move it to that PR?

Reply:
PR was merged, so I have included tests here.